bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,428 @@
+#!/usr/bin/env ruby
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require_relative '../../lib/state_abbr'
+require src_dir 'database'
+require_relative('../../dashboard/config/environment')
+require 'cdo/properties'
+require 'json'
+
+DASHBOARD_REPORTING_DB_READONLY = sequel_connect(
+  CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
+
+UPDATE_BATCH_SIZE = 100
+
+DASHBOARD_DB_NAME = "dashboard_#{Rails.env}"
+
+# List of valid PD courses. If more courses get added, need to update this list and also schema in Pardot. We
+# need to filter here to known courses in the Pardot schema - we can't blindly pass values through
+COURSE_LIST = "'CS Fundamentals','CS in Algebra','CS in Science','CS Principles','Exploring Computer Science'"
+
+# values of forms.kind field we care about
+FORM_KINDS = %w(BringToSchool2013 CSEdWeekEvent2013 DistrictPartnerSubmission HelpUs2013 Petition K5OnlineProfessionalDevelopmentPostSurvey)
+
+# information about presence of which forms submitted by a user get recorded in which
+# rollup field with which value
+FORM_INFOS =
+  [
+    {kind: "'CSEdWeekEvent2013'", dest_field: "hoc_organizer_years", dest_value: "'2013'"},
+    {kind: "'HocSignup2014'", dest_field: "hoc_organizer_years", dest_value: "'2014'"},
+    {kind: "'HocSignup2015'", dest_field: "hoc_organizer_years", dest_value: "'2015'"},
+    {kind: "'HocSignup2016'", dest_field: "hoc_organizer_years", dest_value: "'2016'"},
+    {kind: "'Petition'", dest_field: "roles", dest_value: "'Petition Signer'"}
+  ]
+
+def log(s)
+  puts s      # emit to stdout
```

Comment:
This is because I am going to be running interactively in production to start with before running it literally as a cron job and need to see output when I run interactively. I will remove the stdout output before it turns into cron job in production.